pandas/io/sql.py

```diff
@@ -144,16 +143,15 @@ def _convert_arrays_to_dataframe(
     data,
     columns,
     coerce_float: bool = True,
-    use_nullable_dtypes: bool = False,
+    dtype_backend: DtypeBackend | Literal["numpy"] = "numpy",
```

Comment:
For these internal usages of `dtype_backend`, is there a particular reason why you're using `"numpy"` instead of `no_default`?

Reply:
We are checking dtype_backend == "numpy" quite often in sql and this reads better than checking dtype_backend is/is not lib.no_default. This clutters the if conditions too much imo.

Reply:
Okay that's fine then. Can you make sure we have tests where our public facing function with `dtype_backend="numpy"` raises a `ValueError`?

Reply:
We don't yet. I wanted to a check for all functions in a follow up. This would have made this pr even larger.

Reply:
Sure. Let make sure we have that checking before we release the RC

Reply:
I’ll start a pr that builds on top of this one